Source/WTF/Scripts/audit-spi-if-needed.sh

```diff
@@ -1,46 +1,83 @@
-#!/bin/sh
+#!/bin/bash
+# bash (not sh) is required to support hyphens in function names (the audit-spi
+# fallback case below) and for array variables.
 set -e
 
-program="$(dirname $(dirname $(dirname "${SRCROOT}")))/${WK_ADDITIONAL_SCRIPTS_DIR}/audit-spi"
-[ -f "${program}" ] || program="$(dirname $(dirname "${SRCROOT}"))/Tools/Scripts/audit-spi"
+PATH="${SRCROOT}/../../Tools/Scripts:${PATH}"
+if [ -n "${WK_ADDITIONAL_SCRIPTS_DIR}" ]; then
+    PATH="${SRCROOT}/../../../${WK_ADDITIONAL_SCRIPTS_DIR}:${PATH}"
+fi
+
+if [[ -z "$(type -t audit-spi)" && -d "${WK_WORKSPACE_DIR}/Tools/Scripts/libraries/webkitapipy" ]]; then
+    # Tools/Scripts/audit-spi and its webkitpy dependencies are not submitted
+    # to the production build environment.
+    # In their place, call into the webkitapipy python library directly. This
+    # bypasses the autoinstaller and does not support incremental build
+    # tracking of the sources in WK_ADDITIONAL_SCRIPTS_DIR, so it's only
+    # appropriate for production builds where a minimal version of the tool
+    # will suffice.
+    function audit-spi {
+        PYTHONPATH="${WK_WEBKITADDITIONS_HEADERS_FOLDER_PATH}/Scripts:${WK_WORKSPACE_DIR}/Tools/Scripts/libraries/webkitapipy" python3 -m webkitapipy.program $@
+    }
+fi
 
 # Xcode doesn't expose the name of the discovered dependency file, but by convention, it is
 # the same basename as the timestamp output.
 depfile="${SCRIPT_OUTPUT_FILE_0/%.timestamp/.d}"
 
-eval allowlists=(${WK_AUDIT_SPI_ALLOWLISTS})
+eval FRAMEWORK_SEARCH_PATHS=(${BUILT_PRODUCTS_DIR} ${FRAMEWORK_SEARCH_PATHS} ${SYSTEM_FRAMEWORK_SEARCH_PATHS})
+eval LIBRARY_SEARCH_PATHS=(${BUILT_PRODUCTS_DIR} ${LIBRARY_SEARCH_PATHS} ${SYSTEM_LIBRARY_SEARCH_PATHS})
+eval WK_AUDIT_SPI_ALLOWLISTS=(${WK_AUDIT_SPI_ALLOWLISTS})
 
-# For compatibility with offline build environments.
-export DISABLE_WEBKITCOREPY_AUTOINSTALLER=1
-
-if [[ "${WK_AUDIT_SPI}" == YES && -f "${program}" ]]; then
+if [[ "${WK_AUDIT_SPI}" == YES && -n "$(type -t audit-spi)" && "${ACTION}" != "installhdrs" && "${ACTION}" != "installapi" ]]; then
     mkdir -p "${OBJROOT}/WebKitSDKDBs"
 
-    # WK_SDKDB_DIR is a directory of directories named according to SDK
-    # versions. Pick the versioned directory closest to the active SDK, but not
-    # greater. If all available directories are for newer SDKs, fall back to
-    # the last one.
-    for versioned_sdkdb_dir in $(printf '%s\n' ${WK_SDKDB_DIR}/${PLATFORM_NAME}* | sort -rV); do
-        if printf '%s\n' ${versioned_sdkdb_dir#${WK_SDKDB_DIR}/} ${SDK_NAME%.internal} | sort -CV; then
-            break
-        fi
-    done
+    if [ "${DEPLOYMENT_LOCATION}" = YES ]; then
```

Comment:
nit: I'm using `==` elsewhere, so I should match style, even though this behaves the same.